### ISSUE TITLE
hypothesis used in package code but is only a dev extra, not an installation requirement

### ISSUE THREAD

**Issue by feuGeneA** [ethereum/web3.py#1179]
* Version: 5.0.0-alpha.2
* Python: 3.6
* OS: linux

### What was wrong?

Seems that just importing `utils/datatypes` requires you to have the `hypothesis` package:

```
ImportError while importing test module '/home/circleci/repo/python-packages/order_utils/test/test_asset_data_utils.py'.
Hint: make sure your test modules/packages have valid Python names.
Traceback:
test/test_asset_data_utils.py:5: in <module>
    from zero_ex.order_utils.asset_data_utils import (
src/zero_ex/order_utils/__init__.py:25: in <module>
    from web3.utils import datatypes
/usr/local/lib/python3.7/site-packages/web3/utils/__init__.py:1: in <module>
    from web3._utils import (  # noqa: F401
/usr/local/lib/python3.7/site-packages/web3/_utils/hypothesis.py:1: in <module>
    from hypothesis import (
E   ModuleNotFoundError: No module named 'hypothesis'
```

But the dependency is only listed in `setup.py` under `extras_require['dev']`.

### How can it be fixed?

Move the `hypothesis` dependency from `extras_require` to `install_requires`.

---

If my analysis is correct here, let me know and I'll be happy to submit a PR.

**Comment by pipermerriam** [comment]
Ah, a side-effect of our move of `web3.utils -> web3._utils`!

I believe the appropriate fix for this would be to move `web3._utils.hypothesis` to live under `web3.tools.hypothesis`.

**Comment by Ryan-Gordon** [comment]
Seen this issue and figured I would take a crack at it. 

Moved web3._utils.hypothesis to live under web3.tools.hypothesis per your request @pipermerriam and also updated the __init__.py to reflect the change.

I can now successfully import `from web3.utils import datatypes`

@feuGeneA how were you building the web3 package ? 
I tried 3 ways from source : 

+ `pip install .` -- gathered from docs, error remains as hypothesis isin't installed

However 
+ `pip install ".[tester]"`
and
+ `pip install ".[dev]"`

Work as they both trigger the `extras_require` which installs hypothesis 

Let me know if theres any issues

**Comment by Ryan-Gordon** [comment]
All CI jobs passing 🙌

**Comment by kclowes** [comment]
Old issue cleanup - I believe this was fixed in v5.0.0a8.
